### ISSUE TITLE
Messages appear inside safe area in iPhone X 

### ISSUE THREAD

**Issue by v-ken** [badoo/Chatto#398]
![simulator screen shot - iphone x - 2017-11-08 at 11 27 32](https://user-images.githubusercontent.com/5682879/32525309-f36c7c26-c477-11e7-982f-038b5c7916d5.png)

As you can see from the screenshot of the Chatto demo project above, the messages scrolls below the input toolbar inside the safe area on iPhone X

**Comment by OneSman7** [comment]
They should behave like this.
It would be more appropriate to scale input view container into the safe area and adjust input views to take safe insets into account.

**Comment by guisua** [comment]
I have this exact problem, too. As a temporary solution, this works:

```
// add a property to your view controller

var bottomSafeAreaView: UIView!


// and then override viewDidLayoutSubviews

override func viewDidLayoutSubviews() {
    super.viewDidLayoutSubviews()
    if #available(iOS 11.0, *) {
        if bottomSafeAreaView == nil {
            let safeAreaHeight = self.view.safeAreaInsets.bottom
            bottomSafeAreaView = UIView(frame: CGRect(x: 0, y: self.view.frame.height-safeAreaHeight, width: self.view.frame.width, height: safeAreaHeight))
            bottomSafeAreaView.backgroundColor = view.backgroundColor
            self.view.addSubview(bottomSafeAreaView)
        }
    }
}
```
